src/client.rs

```diff
@@ -1,4 +1,156 @@
 //! Client implementation of the HTTP/2.0 protocol.
+//!
+//! # Getting started
+//!
+//! Running an HTTP/2.0 client requires the caller to establish the underlying
+//! connection as well as get the connection to a state that is ready to begin
+//! the HTTP/2.0 handshake. See [here](../index.html#handshake) for more
+//! details.
+//!
+//! This could be as basic as using Tokio's [`TcpStream`] to connect to a remote
+//! host, but usually it means using either ALPN or HTTP/1.1 protocol upgrades.
+//!
+//! Once a connection is obtained, it is passed to [`handshake`], which will
+//! begin the [HTTP/2.0 handshake]. This returns a future that completes once
+//! the handshake process is performed and HTTP/2.0 streams may be initialized.
+//!
+//! [`handshake`] uses default configuration values. THere are a number of
+//! settings that can be changed by using [`Builder`] instead.
+//!
+//! Once the the handshake future completes, the caller is provided with a
+//! [`Connection`] instance and a [`SendRequest`] instance. The [`Connection`]
+//! instance is used to drive the connection (see [Managing the connection]).
+//! The [`SendRequest`] instance is used to initialize new streams (see [Making
+//! requests]).
+//!
+//! # Making requests
+//!
+//! Requests are made using the [`SendRequest`] handle provided by the handshake
+//! future. Once the request is submitted, an HTTP/2.0 stream is initialized and
+//! the request is sent to the server.
+//!
+//! A request body and request trailers are sent using [`SendRequest`] and the
+//! server's response is returned once the [`ResponseFuture`] future completes.
+//! Both the [`SendStream`] and [`ResponseFuture`] instances are returned by
+//! [`SendRequest::send_request`] and are tied to the HTTP/2.0 stream
+//! initialized by the sent request.
+//!
+//! The [`MAX_CONCURRENT_STREAMS`] setting is enforced by [`SendRequest`]. A
+//! request cannot be sent if the number of currently active streams has reached
+//! the maximum permitted for the connection.
```

Comment:
Does this result in a UserError?  Is the user responsible for ensuring this or does the cilent enforce this concurrency with buffering?

My point is _A request cannot be sent_ is not all that clear, whereas _Users must not send_ etc, would be a bit clearer to me.

Reply:
Perhaps this is all clearer if this note comes after the `poll_ready` section?